packages/next/build/babel/loader/get-config.ts

```diff
@@ -0,0 +1,193 @@
+  loaderOptions: NextBabelLoaderOptions,
+  source: string,
+  filename: string
+) {
+  const { hasReactRefresh, isServer, development, pagesDir } = loaderOptions
+  const isPageFile = filename.startsWith(pagesDir)
+
+  const applyCommonJsItem =
+    source.indexOf('module.exports') === -1
+      ? null
+      : createConfigItem(require('../plugins/commonjs'), { type: 'plugin' })
+  const reactRefreshItem = hasReactRefresh
+    ? createConfigItem(
+        [require('react-refresh/babel'), { skipEnvCheck: true }],
+        { type: 'plugin' }
+      )
+    : null
+  const noAnonymousDefaultExportItem =
+    hasReactRefresh && !isServer
+      ? createConfigItem(
+          [require('../plugins/no-anonymous-default-export'), {}],
+          { type: 'plugin' }
+        )
+      : null
+  const pageConfigItem =
+    !isServer && isPageFile
+      ? createConfigItem([require('../plugins/next-page-config')], {
+          type: 'plugin',
+        })
+      : null
+  const disallowExportAllItem =
+    !isServer && isPageFile
+      ? createConfigItem(
+          [require('../plugins/next-page-disallow-re-export-all-exports')],
+          { type: 'plugin' }
+        )
+      : null
+  const transformDefineItem = createConfigItem(
+    [
+      require.resolve('next/dist/compiled/babel/plugin-transform-define'),
+      {
+        'process.env.NODE_ENV': development ? 'development' : 'production',
+        'typeof window': isServer ? 'undefined' : 'object',
+        'process.browser': isServer ? false : true,
+      },
+      'next-js-transform-define-instance',
+    ],
+    { type: 'plugin' }
+  )
+  const nextSsgItem =
+    !isServer && isPageFile
+      ? createConfigItem([require.resolve('../plugins/next-ssg-transform')], {
+          type: 'plugin',
+        })
+      : null
+
+  return [
+    noAnonymousDefaultExportItem,
+    reactRefreshItem,
+    pageConfigItem,
+    disallowExportAllItem,
+    applyCommonJsItem,
+    transformDefineItem,
+    nextSsgItem,
+  ].filter(Boolean)
+}
+
+function getOverrides(overrides = []) {
+  const commonJsItem = createConfigItem(
+    require('next/dist/compiled/babel/plugin-transform-modules-commonjs'),
+    { type: 'plugin' }
+  )
+
+  return [
+    ...overrides,
+    {
+      test: [
+        /next[\\/]dist[\\/]next-server[\\/]lib/,
+        /next[\\/]dist[\\/]client/,
+        /next[\\/]dist[\\/]pages/,
+      ],
+      plugins: [commonJsItem],
+    },
+  ]
+}
+
+const configs = new Map()
+export default function getConfig(
+  this: NextJsLoaderContext,
+  {
+    source,
+    loaderOptions,
+    inputSourceMap,
+    target,
+    filename,
+  }: {
+    source: string
+    loaderOptions: NextBabelLoaderOptions
+    inputSourceMap?: object | null
+    target: string
+    filename: string
+  }
+) {
+  const {
+    presets = [],
+    isServer,
+    pagesDir,
+    development,
+    hasReactRefresh,
+    hasJsxRuntime,
+  } = loaderOptions
+  const configKey = `${isServer ? 'server' : 'client'}:${filename}`
```

Comment:
This is going to keep increasing the `configs` map and as such increase memory usage right 🤔

Reply:
Yes. It seemed like this cost would be negligible, but I'll look at whether there is a closure captured in the config somewhere that would cause a memory leak.